src/services/edge-application-rules-engine-services/edit-rules-engine-service.js

```diff
@@ -0,0 +1,90 @@
+import { AxiosHttpClientAdapter } from '@/services/axios/AxiosHttpClientAdapter'
+import { makeEdgeApplicationBaseUrl } from '../edge-application-services/make-edge-application-base-url'
+import * as Errors from '@/services/axios/errors'
+
+export const editRulesEngineService = async ({ id, payload, reorder = false }) => {
+  const parsedPayload = adapt(payload, reorder)
+
+  const rulesPhase =
+    payload.phase.content === 'Default' ? 'request' : payload.phase.content?.toLowerCase()
+
+  let httpResponse = await AxiosHttpClientAdapter.request({
+    url: `${makeEdgeApplicationBaseUrl()}/${id}/rules_engine/${rulesPhase}/rules/${payload.id}`,
+    method: 'PATCH',
+    body: parsedPayload
+  })
+
+  return parseHttpResponse(httpResponse)
+}
+
+const adapt = (payload, reorder) => {
+  let parsedPaylod
+
+  if (!reorder) {
+    const { name, phase, behaviors, criteria, isActive, description } = payload
+    parsedPaylod = {
+      name,
+      phase: phase.content,
+      behaviors,
+      criteria,
+      is_active: isActive,
+      description
+    }
+  } else {
+    parsedPaylod = {
+      order: parseInt(payload.order)
+    }
+  }
+
+  return parsedPaylod
+}
+
+/**
+ * @param {Object} errorSchema - The error schema.
+ * @param {string} key - The error key of error schema.
+ * @returns {string|undefined} The result message based on the status code.
+ */
+const extractErrorKey = (errorSchema, key) => {
+  if (Array.isArray(errorSchema[key])) {
+    return errorSchema[key]?.[0]
+  }
+  return errorSchema[key]
+}
+
+/**
+ * @param {Object} httpResponse - The HTTP response object.
+ * @param {Object} httpResponse.body - The response body.
```

Comment:
@param {Object} httpResponse.body - The response body.
para
@param {Array<Object>} httpResponse.body - The response body.